src/native/crt.c

```diff
@@ -654,6 +657,30 @@ void JNICALL Java_software_amazon_awssdk_crt_CRT_dumpNativeMemory(JNIEnv *env, j
     }
 }
 
+/* FIPS mode can be checked if OpenSSL was configured and built for FIPS which then defines OPENSSL_FIPS.
+ *
+ * AWS-LC always defines FIPS_mode() that you can call and check what the library was built with. It does not define
+ * a public OPENSSL_FIPS/AWSLC_FIPS macro that we can (or need to) check here
+ *
+ * Safeguard with macro's, for example because Libressl dosn't define
+ * FIPS_mode() by default.
+ *
+ * Note: FIPS_mode() does not change the FIPS state of libcrypto. This only returns the current state. Applications
+ * using s2n must call FIPS_mode_set(1) prior to s2n_init.
+ * */
+JNIEXPORT
+jboolean JNICALL Java_software_amazon_awssdk_crt_CRT_libcryptoIsFIPS(JNIEnv *env, jclass jni_crt_class) {
+    (void)env;
```

Comment:
> must call FIPS_mode_set(1) prior to s2n_init

Is the CRT calling `FIPS_mode_set(1)` anywhere? Is there anything else we need to do to ensure S2N is doing everything the FIPS way?

Should we add an is_fips function to aws-c-io? aws-c-cal?

Should we simply call this function "isFips" instead of "libcryptoIsFips"?

Reply:
> Is the CRT calling FIPS_mode_set(1) anywhere? Is there anything else we need to do to ensure S2N is doing everything the FIPS way?

For java, we only support aws-lc, and aws-lc only rely on built-time to set the FIPS mode, [here](https://github.com/aws/aws-lc/blob/6d9a7675d1f7c2f159b76e4ade309220d54d96f4/include/openssl/crypto.h#L183-L185). So, we don't need to call `FIPS_mode_set(1)`. I copy/paste the part from s2n. I will remove this. 

s2n set up fips mode when libcrypto has FIPS mode enabled. [here](https://github.com/aws/s2n-tls/blob/main/crypto/s2n_fips.c#L51-L52). But, to really make sure we do the FIPS way, it's still the user's responsibility to use the right cypher suits, which is designed by s2n and aws-lc.

Discussed offline, for the possibility that we may support openssl in the future, changed the impl

> Should we add an is_fips function to aws-c-io? aws-c-cal?

probably not, they have access to `openssl/crypto.h`